Sources/NIO/Bootstrap.swift

```diff
@@ -205,18 +205,22 @@ public final class ServerBootstrap {
             return eventLoop.newFailedFuture(error: error)
         }
 
-        return serverChannelInit(serverChannel).then {
-            serverChannel.pipeline.add(handler: AcceptHandler(childChannelInitializer: childChannelInit,
-                                                              childChannelOptions: childChannelOptions))
-        }.then {
-            serverChannelOptions.applyAll(channel: serverChannel)
+        return eventLoop.submit {
+            return serverChannelInit(serverChannel).then {
+                serverChannel.pipeline.add(handler: AcceptHandler(childChannelInitializer: childChannelInit,
+                                                                  childChannelOptions: childChannelOptions))
+            }.then {
+                serverChannelOptions.applyAll(channel: serverChannel)
+            }.then {
+                register(eventLoop, serverChannel)
+            }.map {
+                serverChannel as Channel
```

Comment:
@weissi why we need this ?

Reply:
@normanmaurer because otherwise the compiler complains can't do `EventLoopFuture<ServerSocketChannel>` as `EventLoopFuture<Channel>`